solr/modules/langid/src/test/org/apache/solr/update/processor/LanguageIdentifierUpdateProcessorFactoryTestCase.java

```diff
@@ -464,6 +466,101 @@ public void testMapIndividual() throws Exception {
     assertTrue(mappedIndividual.containsKey("text2_ru"));
   }
 
+  @Test
+  public void testAllowlistWithFallback() throws Exception {
+    ModifiableSolrParams parameters = new ModifiableSolrParams();
+    parameters.add("langid.fl", "name,subject");
+    parameters.add("langid.langField", "language_s");
+    parameters.add("langid.allowlist", "no,en ,,,sv"); // Any other language will fallback to EN
+    parameters.add("langid.fallback", "en");
+    liProcessor = createLangIdProcessor(parameters);
+
+    // Make sure that empty language codes have been filtered out and others trimmed.
```

Comment:
Should be no need to do an actual detection here (which currently fails tests). Suffice to check that the allowList got populated:
```suggestion
    assertEquals(Set.of("no", "en", "sv"), liProcessor.langAllowlist);
```